tests/test_json.py

```
@@ -875,6 +875,49 @@ def test_jsanitize(self):
+        skipped_dict = jsanitize(composite_obj, strict="skip", recursive_msonable=True)
+        assert skipped_dict["name"] == "test", "String values should remain unchanged"
+        assert (
+            skipped_dict["unserializable"]._a == 1
+        ), "Unserializable object should be preserved in skip mode"
+        assert (
+            skipped_dict["msonable"]["a"] == 1
+        ), "MSONable object should be properly serialized"
+
+        # Test non-strict mode converts unserializable to string
+        assert isinstance(
+            dict_with_str["unserializable"], str
```

Ruff (0.8.2)

Undefined name `dict_with_str`

(F821)